File: C/bellman-ford/meta-llama-4-scout-17b-16e-instruct.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

typedef struct Edge {
    int src, dest, weight;
} Edge;

typedef struct Graph {
    int V, E;
    Edge* edges;
} Graph;

Graph* createGraph(int V, int E) {
    Graph* graph = (Graph*)malloc(sizeof(Graph));
    graph->V = V;
    graph->E = E;
    graph->edges = (Edge*)malloc(E * sizeof(Edge));
    return graph;
}

int bellmanFord(Graph* graph, int src) {
    int* dist = (int*)malloc(graph->V * sizeof(int));
    for (int i = 0; i < graph->V; i++) {
        dist[i] = INT_MAX;
    }
    dist[src] = 0;

    for (int _ = 0; _ < graph->V - 1; _++) {
        for (int i = 0; i < graph->E; i++) {
            int u = graph->edges[i].src;
            int v = graph->edges[i].dest;
            int w = graph->edges[i].weight;
            if (dist[u] != INT_MAX && dist[u] + w < dist[v]) {
                dist[v] = dist[u] + w;
            }
        }
    }

    for (int i = 0; i < graph->E; i++) {
        int u = graph->edges[i].src;
        int v = graph->edges[i].dest;
        int w = graph->edges[i].weight;
        if (dist[u] != INT_MAX && dist[u] + w < dist[v]) {
            return -1; // Negative cycle detected
        }
    }

    free(dist);
    return 0; // No negative cycle detected
}

int main() {
    FILE* file = fopen("../../graph_input.txt", "r");
    if (!file) {
        return 1;
    }

    int V, E;
    if (fscanf(file, "%d %d", &V, &E) != 2) {
        fclose(file);
        return 1;
    }

    Graph* graph = createGraph(V, E);
    if (!graph || !graph->edges) {
        fclose(file);
        return 1;
    }

    for (int i = 0; i < E; i++) {
        if (fscanf(file, "%d %d %d", &graph->edges[i].src, &graph->edges[i].dest, &graph->edges[i].weight) != 3) {
            free(graph->edges);
            free(graph);
            fclose(file);
            return 1;
        }
    }
    fclose(file);

    int result = bellmanFord(graph, 0);

    free(graph->edges);
    free(graph);

    return 0;
}

File: Java/bellman-ford/metaLlama4Scout17b16eInstruct.java
import java.util.Random;

public class MetaLlama4Scout17b16eInstruct {
    static class Edge {
        int src, dest, weight;
    }

    static class Graph {
        int V;
        Edge[] edge;
        int E;

        public Graph(int vertices) {
            V = vertices;
            edge = new Edge[vertices * (vertices - 1)];
            E = 0;
        }

        public void addEdge(int src, int dest, int weight) {
            edge[E] = new Edge();
            edge[E].src = src;
            edge[E].dest = dest;
            edge[E].weight = weight;
            E++;
        }

        public boolean bellmanFord(int src) {
            int[] dist = new int[V];
            for (int i = 0; i < V; i++)
                dist[i] = Integer.MAX_VALUE;
            dist[src] = 0;

            for (int i = 0; i < V - 1; i++) {
                for (int j = 0; j < E; j++) {
                    int u = edge[j].src;
                    int v = edge[j].dest;
                    int w = edge[j].weight;
                    if (dist[u] != Integer.MAX_VALUE && dist[u] + w < dist[v])
                        dist[v] = dist[u] + w;
                }
            }

            for (int i = 0; i < E; i++) {
                int u = edge[i].src;
                int v = edge[i].dest;
                int w = edge[i].weight;
                if (dist[u] != Integer.MAX_VALUE && dist[u] + w < dist[v])
                    return false;
            }
            return true;
        }
    }

    public static void main(String[] args) {
        final int V = 1000; // Número de vértices
        Graph g = new Graph(V);

        // Lê as arestas do arquivo
        try (java.io.BufferedReader br = new java.io.BufferedReader(new java.io.FileReader("../../graph_input.txt"))) {
            String line;
            while ((line = br.readLine()) != null) {
                String[] parts = line.trim().split("\\s+");
                if (parts.length != 3) continue;

                int src = Integer.parseInt(parts[0]);
                int dest = Integer.parseInt(parts[1]);
                int weight = Integer.parseInt(parts[2]);

                g.addEdge(src, dest, weight);
            }
        } catch (java.io.IOException e) {
            return;
        }

        boolean hasNoNegativeCycle = g.bellmanFord(0);
    }

}